### DISCUSSION TITLE
How to restore ability to use middle mouse button (on X11) to paste selected text?

### DISCUSSION THREAD

**Discussion by Andrew15-5** [zellij-org/zellij#3017] [Q&A] [answered]
If I just open my terminal with my shell, then I can paste selected text from a browser. If I instead open the same terminal, but with zellij and inside of it my shell, then the MMB paste action become unavailable for some reason.

When I check the dumped default config, I have found something similar:

```
// Toggle enabling the mouse mode.
// On certain configurations, or terminals this could
// potentially interfere with copying text.
// Options:
//   - true (default)
//   - false
//
// mouse_mode false
```

But since it's already enabled by default, it means that this doesn't control what I want to do.

**Comment by Lockszmith-GH** [comment]
Have you tried `Ctrl`+`G` and then pasting?

**Reply by Andrew15-5** [reply]
Yes. I found out that <kbd>Ctrl</kbd> + <kbd>/</kbd> also doesn't work. If this program is intended for daily local host use, then this is very limiting, in a way. I hope there is a way to make it work, or a new feature request can be created.

**Reply by Lockszmith-GH** [reply]
I personally use the keyboard more than the mouse

`Ctrl`+`Shift`+`V` is usually what I use to paste.

The reason you are experiencing this change in behavior is because zellij takes over mouse control.

This allows selecting text within a pane, instead of lines being absolute to the try connection.

I think you can disable mouse control in the config.

**Reply by Andrew15-5** [reply, marked as answer]
> The reason you are experiencing this change in behavior is because zellij takes over mouse control.

I also wanted to add that with default Neovim I can use MMB to paste. So how come Neovim inside zellij can take over mouse control from zellij?

Uncommenting `mouse_mode false` does solve the problem, and I don't really plan on using cursor to interact with zellij, so this is good enough for me.

> I personally use the keyboard more than the mouse

And I use the keyboard so much that I hate grabbing the mouse every single time, so I have a touchpad (and mouse is for special cases).

 Sometimes it's just easier to select text from GUI with touchpad and then paste it with 3-finger tap to a terminal. I don't want to over-pollute my clipboard manager's history, so I don't want to constantly copy  everything to the clipboard (instead selection and MMB use primary selection (default in `xclip`, i.e., `XA_PRIMARY`).

**Reply by Lockszmith-GH** [reply]
nvim has their own mouse capturing. I've noticed this when selecting within nvim doesn't end up in the clipboard (at least not on Windows).

There I use `Shift` while selecting, this bypasses both zellij's and nvim's mouse capturing. But that might be a feature of the terminal I'm using - it works the same for me in Tabby and Windows Terminal.

I'm glad that `mouse_mode false` solved the current problem for you.

**Comment by xbello** [comment]
I also couldn't do proper MMB paste, specially when selecting multiline, that with `mouse_mode false` spans the selection to the adjacent panels, but with `mouse_mode true` doesn`t paste on MMB. My workaround:

1. Install both `xbindkeys` and `xte`.

2. Create a $HOME/xbindkeysrc with `xbindkeys --defaults > $HOME/.xbindkeyrc`.

3. Link the MMB click to "Ctrl + Shift + V" adding the following two lines to `.xbindkeyrc`, including quotes:

       "xte 'keydown Control_L' 'keydown Shift_L' 'key V' 'keyup Shift_L' 'keyup Control_L'"
       b:2

It seems that Zellij copies-on-select, but not to the buffer that MMB uses, but to the buffer that is retrieved with Ctl+Shf+V

**Reply by Andrew15-5** [reply]
MMB paste works if you hold shift key. I discovered this by accident. Either just by myself, or by reading the docs in the default config file, or by reading docs on the website. 

So now I can have this feature on and be able to paste text. The downside is that you can't really do this with one hand.

**Reply by morgoth** [reply]
@Andrew15-5 Thanks for the tip.
@imsnif Is it possible to do a custom keybinding for middle mouse to this paste action? Since it's working with "Shift + MMB", I guess it should be possible to bind MMB itself, but I cannot find such action and MMB in the https://github.com/zellij-org/zellij/blob/main/zellij-utils/assets/config/default.kdl nor https://zellij.dev/documentation/keybindings-possible-actions

**Reply by ManuLinares** [reply]
Have you found a solution to this?

I want mouse mode enabled, and rebind the "middle click" to paste.

**Reply by Lockszmith-GH** [reply]
What platform are you on? @ManuLinares 
As in - the platform your keyboard and mouse are connected to - not necessarily the platform Zellij is running on.

**Reply by ManuLinares** [reply]
> What platform are you on? @ManuLinares As in - the platform your keyboard and mouse are connected to - not necessarily the platform Zellij is running on.

I'm runnig archlinux, XFCE. Alacritty -> zellij

When I select a text on some other app, I can paste to zellij with "Shift + middle_click". I want to just "middle_click" to paste that buffer.

**Reply by Lockszmith-GH** [reply]
Not sure how to help here, if you've tried what xbello suggested and it still fails you.

Does middle click paste work when mouse_mode is off?

**Reply by ManuLinares** [reply]
> Does middle click paste work when mouse_mode is off?

Yes, when mouse_mode is off, middle_click paste works. 
With mouse_mode on, only shift+middle_click paste work.

Any way to remap middle_click?
